tests/test_bcrypt.py

```diff
@@ -464,6 +467,7 @@ def test_kdf_no_warn_rounds():
     bcrypt.kdf(b"password", b"salt", 10, 10, True)
 
 
+@pytest.mark.thread_unsafe()
```

Comment:
Why is this thread unsafe?

Reply:
It's because `warnings.catch_warnings` [is thread-unsafe](https://docs.python.org/3/library/warnings.html#warnings.catch_warnings) so `pytest.warns` is as well. There's work [actively going on](https://github.com/python/cpython/issues/128384) in CPython to (hopefully) provide a way forward on this, it's by far the most common source of thread safety issues in Python test suites when running tests in multiple threads.

Reply:
Sigh. Would you mind adding a comment to each of these with the reason (just a single sentence: "warning state is global" is fine).